test/Common.test.cpp

```diff
@@ -0,0 +1,8 @@
+#include "NAS2D/Common.h"
+#include <gtest/gtest.h>
+#include <gmock/gmock.h>
+
+
+TEST(Version, versionString) {
+	EXPECT_THAT(NAS2D::versionString(), testing::MatchesRegex("[[:digit:]]+\\.[[:digit:]]+\\.[[:digit:]]+"));
```

Comment:
Using raw-string-literals will make the regex more clear and remove the need to escape the slashes: `R"([[:digit:]]+\.[[:digit:]]+\.[[:digit:]]+)"`

Reply:
Ahh yes, I tried [raw string literals](https://en.cppreference.com/w/cpp/language/string_literal), but was stumbling with the syntax. I didn't think to try enclosing the string with parenthesis while using an empty delimiter.

What you posted works. The empty delimiter has me wondering though, what happens if someone inserts parenthesis into the regex for a capture group. Could that end the raw string literal early? I suppose it might also require a quote. It's not too clear to me how the new rules interact.

Reply:
As long as the sequence `)"` does not exist within the string it will work. In cases where it does you can add a delimiter you know won't appear:

    std::string testA = R"(Hello World)";
    std::string testB = R"("C(A|B)")";  <-- this fails and ends early.
    std::string testC = R"delim("C(A|B)")delim";  <-- this works because the delimiter is now )delim"